### PR TITLE
Checking that startDate is before endDate

### PR DESCRIPTION
**Pull request by alex-deaconu**
Conditionally use locale for date/time saving

### CODE REVIEW

File: src/timeline/dtv-datepicker-localdate.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+              if (!viewValue) {
+                return null;
+              }
+
+              if (useLocaldate) {
+                // undo the timezone adjustment we did during the formatting
+                viewValue.setMinutes(viewValue.getMinutes() - viewValue.getTimezoneOffset());
+              } else {
+                viewValue = formatDate(viewValue);
+              }
```
Comment (alex-deaconu) [inline src/timeline/dtv-datepicker-localdate.js]: Either remove timezone offset from the Date, or output it in a "timezone-less" format...

File: src/timeline/dtv-datepicker-localdate.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+              if (!modelValue) {
+                return undefined;
+              }
+              // date constructor will apply timezone deviations from UTC (i.e. if locale is behind UTC 'dt' will be one day behind)
+              var dt = new Date(modelValue);
+
+              if (useLocaldate) {
+                // 'undo' the timezone offset again (so we end up on the original date again)
+                dt.setMinutes(dt.getMinutes() + dt.getTimezoneOffset());
+              }
```
Comment (alex-deaconu) [inline src/timeline/dtv-datepicker-localdate.js]: Don't remove offset if it's not actually there (assume date doesn't have it if `useLocaldate = false`).

File: src/timeline/svc-timeline-factory.js [hunk 3]
```diff
@@ -118,7 +117,7 @@ angular.module("risevision.common.components.timeline.services")
 
         var _init = function () {
           _timeline.timeDefined = _timeline.timeDefined || false;
-          _timeline.startDate = _timeline.startDate || new Date();
+          _timeline.startDate = _timeline.startDate || _getDateTime(0, 0);
```
Comment (alex-deaconu) [inline src/timeline/svc-timeline-factory.js]: Get a date where the hour is 12:00AM.

File: src/timeline/svc-timeline-factory.js [hunk 4]
```diff
@@ -11,12 +11,11 @@ angular.module("risevision.common.components.timeline.services")
...
         };
-        var _getDateTime = function (hours, minutes) {
+        var _getDateTime = function (hour, minute) {
           var d = new Date();
-          d.setHours(hours);
-          d.setMinutes(minutes);
 
-          return d;
+          return new Date(Date.UTC(d.getFullYear(), d.getMonth(), d.getDate(),
+            hour, minute, 0));
```
Comment (alex-deaconu) [inline src/timeline/svc-timeline-factory.js]: Fix issue where milliseconds were kept on the date.

**Comment by alex-deaconu** [comment]
I've annotated the code above with more details on the implementation. Let me know if you have any questions.

@rodrigopavezi please review. Thanks!

**Comment by alex-deaconu** [comment]
@rodrigopavezi see my latest commit. This fixes issues with initialization of the Dates for both the Schedule and Schedule items.

Please review. Thanks!

**Comment by rodrigopavezi** [comment]
LG :+1:

**Pull request merged**